lib.php

```diff
@@ -1442,6 +1490,12 @@ function facetoface_write_activity_attendance(&$worksheet, $startingrow, $faceto
                 $signup->grade = '-';
             }
 
+            $stream = get_user_stream($userid);
+            $signup->stream = !empty($stream) ? $stream : '-';
```

Comment:
This isn't strictly something that needs to change but I figured since your functions are always returning null when there's no value you might as well make use of the null coalescing operator.
If you've never used it before, this line could be simplified to `$signup->stream = $stream ?? '-'`. This is _effectively_ short-hand for `$signup->stream = (isset($stream) && $stream !== null ? $stream : '-')`.
You also have the 'Elvis operator' which is basically just short hand for when you have a ternary like `$signup->stream = $stream ? $stream : '-'`. Using the elvis operator, this can be shortened to `$signup->stream = $stream ?: '-'`.

Reply:
This is really helpful. Thank you.